app/react/components/buttons/LoadingButton.tsx

```diff
@@ -21,19 +25,22 @@ export function LoadingButton({
       {...buttonProps}
       type={type}
       disabled={disabled || isLoading}
+      icon={LoadingButtonIcon(isLoading)}
     >
-      {isLoading ? (
-        <>
-          <i
-            className="fa fa-circle-notch fa-spin space-right"
-            aria-label="loading"
-            aria-hidden="true"
-          />
-          {loadingText}
-        </>
-      ) : (
-        children
-      )}
+      {isLoading ? loadingText : children}
     </Button>
   );
 }
+
+function LoadingButtonIcon(isLoading: boolean) {
```

Comment:
should take care of the possibility that a default icon is passed. i.e we use
`<LoadingButton icon={Settings}>Submit</LoadingButton>`
```suggestion
function LoadingButtonIcon(isLoading: boolean, defaultIcon: ReactNode) {
```

Reply:
updated